lib/mapped.js

```diff
@@ -30,9 +30,9 @@ MappedCoverage.prototype.addStatement = function (loc, hits) {
         meta = this.meta,
         index = meta.seen[key];
 
-    if (!index) {
-        meta.last.s += 1;
+    if (index === undefined) {
         index = meta.last.s;
+        meta.last.s += 1;
```

Comment:
@bcoe no, the main issue was that this code started from `1` when filling out statement, function, and branch stats, rather than `0` we switched to in https://github.com/istanbuljs/istanbul-lib-instrument/pull/22. Then, when trying to merge stats generated by this code (`1`-based) with ones instrumented by `instanbul-lib-instrument` (`0`-based), it throwed because `0` element was undefined on the first.